libraries/chain/database.cpp

```diff
@@ -2289,15 +2290,25 @@ namespace golos { namespace chain {
                                 for (auto& dvir : itr->delegator_vote_interest_rates) {
                                     auto delegator_claim = claim * dvir.interest_rate / STEEMIT_100_PERCENT;
 
+                                    claim -= delegator_claim;
```

Comment:
You must not change `claim`, it used in `delegator_claim` calculations.
In case of direct rewarding delegators (without adding to existing delegation), you must use 2 separate claim variables: one to reward curator and another to calculate `delegator_claim`